Codes/Brawlback/Brawlback.h

```diff
@@ -57,114 +51,81 @@ inline void updateGamePadSystem() { updateGame(PAD_SYSTEM); }
-        stickY = 0;
-        cStickX = 0;
-        cStickY = 0;
-        LTrigger = 0;
-        RTrigger = 0;
+struct BrawlbackPadImpl {
+    BrawlbackPad _brawlbackPad;
+
+    BrawlbackPadImpl()
+    {
+        _brawlbackPad.buttons = 0;
+        _brawlbackPad.stickX = 0;
+        _brawlbackPad.stickY = 0;
+        _brawlbackPad.cStickX = 0;
+        _brawlbackPad.cStickY = 0;
+        _brawlbackPad.LTrigger = 0;
+        _brawlbackPad.RTrigger = 0;
     }
 };
-#pragma pack(pop)
 
-#pragma pack(push, 4)
+struct PlayerFrameDataImpl {
+    PlayerFrameData _playerFrameData;
 
-struct PlayerFrameData {
-    u32 frame;
-    u8 playerIdx;
-    BrawlbackPad pad;
-
-    PlayerFrameData() {
-        frame = 0;
-        playerIdx = 0;
-        pad = BrawlbackPad();
+    // do these impact the size of the struct?
+    // wouldn't the vtable ptr screw with it being interpreted on gameside???
+    // (since the gameside structs don't have these ctors)
+    PlayerFrameDataImpl() {
+        _playerFrameData.frame = 0;
+        _playerFrameData.playerIdx = 0;
+        _playerFrameData.pad = BrawlbackPadImpl()._brawlbackPad;
     }
-    PlayerFrameData(u32 frame, u8 playerIdx) {
-        frame = frame;
-        playerIdx = playerIdx;
-        pad = BrawlbackPad();
+    PlayerFrameDataImpl(u32 _frame, u8 _playerIdx)
+    {
+        _playerFrameData.frame = _frame;
+        _playerFrameData.playerIdx = _playerIdx;
+        _playerFrameData.pad = BrawlbackPadImpl()._brawlbackPad;
     }
 };
 
-struct FrameData {
-    u32 randomSeed;
-    PlayerFrameData playerFrameDatas[MAX_NUM_PLAYERS];
-    FrameData() {
-        randomSeed = 0;
+struct FrameDataImpl {
+    FrameData _frameData;
+
+    FrameDataImpl()
+    {
+        _frameData.randomSeed = 0;
         for (int i = 0; i < MAX_NUM_PLAYERS; i++) {
-            playerFrameDatas[i] = PlayerFrameData();
+            _frameData.playerFrameDatas[i] = PlayerFrameDataImpl()._playerFrameData;
         }
     }
-    FrameData(u32 frame) {
-        randomSeed = 0;
+    FrameDataImpl(u32 frame)
+    {
+        _frameData.randomSeed = 0;
         for (u8 i = 0; i < MAX_NUM_PLAYERS; i++) {
-            playerFrameDatas[i] = PlayerFrameData(frame, i);
+            _frameData.playerFrameDatas[i] = PlayerFrameDataImpl(frame, i)._playerFrameData;
         }
     }
 };
 
-struct RollbackInfo {
-    bool isUsingPredictedInputs;
-    u32 beginFrame; // frame we realized we have no remote inputs
-    u32 endFrame; // frame we received new remote inputs, and should now resim with those
-    FrameData predictedInputs;
-
-    bool pastFrameDataPopulated;
-    FrameData pastFrameDatas[MAX_ROLLBACK_FRAMES];
-
-    bool hasPreserveBlocks;
+struct RollbackInfoImpl {
+    RollbackInfo _rollbackInfo;
     //vector<PreserveBlock> preserveBlocks;
 };
 
-#pragma pack(pop)
-
-void fillOutGameSettings(GameSettings* settings);
-void MergeGameSettingsIntoGame(GameSettings* settings);
+void fillOutGameSettings(GameSettingsImpl* settings);
```

Comment:
Maybe a bigger comment that's out of scope for this particular review, but a lot of places where we are using pointers, it would make more sense to use references instead.